core/parser/src/main/java/com/blazebit/persistence/parser/expression/CombiningResultVisitorAdapter.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright 2014 - 2017 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.parser.expression;
+
+import com.blazebit.persistence.parser.predicate.BetweenPredicate;
+import com.blazebit.persistence.parser.predicate.BinaryExpressionPredicate;
+import com.blazebit.persistence.parser.predicate.BooleanLiteral;
+import com.blazebit.persistence.parser.predicate.CompoundPredicate;
+import com.blazebit.persistence.parser.predicate.EqPredicate;
+import com.blazebit.persistence.parser.predicate.ExistsPredicate;
+import com.blazebit.persistence.parser.predicate.GePredicate;
+import com.blazebit.persistence.parser.predicate.GtPredicate;
+import com.blazebit.persistence.parser.predicate.InPredicate;
+import com.blazebit.persistence.parser.predicate.IsEmptyPredicate;
+import com.blazebit.persistence.parser.predicate.IsNullPredicate;
+import com.blazebit.persistence.parser.predicate.LePredicate;
+import com.blazebit.persistence.parser.predicate.LikePredicate;
+import com.blazebit.persistence.parser.predicate.LtPredicate;
+import com.blazebit.persistence.parser.predicate.MemberOfPredicate;
+import com.blazebit.persistence.parser.predicate.Predicate;
+
+import java.util.List;
+
+/**
+ * @author Moritz Becker
+ * @since 1.2.0
+ */
+public abstract class CombiningResultVisitorAdapter<T> implements Expression.ResultVisitor<T> {
```

Comment:
Although the idea of a combining visitor is interesting, it is IMO inappropriate for the size detection as the detection can stop traversing the expression tree as soon as a size expression was found, whereas the combining visitor will still traverse the whole tree. Please stick with the AbortableVisitorAdapter.